### DISCUSSION TITLE
How to prevent locale sub-paths for locales with custom domains?

### DISCUSSION THREAD

**Discussion by zadama** [vercel/community#641] [Help]
What can I do to prevent Next/Vercel from creating sub-path pages for locales with custom domains? 

I have two locales in my project: en and es, each with a separate domain, myDomain.com and myDomain.es. Since both of these locales have their own domain, I would like to prevent Next/Vercel from creating pages such as myDomain.com/es or myDomain.es/en. The main reason I want to prevent this is because it will hurt SEO since the same content will appear in two places rather than just one.

**Comment by Babbili** [comment]
@zadama for the SEO, to prevent Search Engine Crawlers from crawling duplicate pages. add in the `head` tag in both of your duplicate pages a `rel='cononical'` `link` tag as follow
```
<link rel="canonical" href="https://myDomain.com"/>
```
This will specify one URL `https://myDomain.com`  as the *canonical version* to crawl for Duplicate pages, see more in Google Search Central [canonical URL](https://developers.google.com/search/docs/advanced/crawling/consolidate-duplicate-urls)

For the i18n Domain Routing, try this configuration
```
// next.config.js
module.exports = {
  i18n: {
    locales: ['en', 'es'],
    defaultLocale: 'en',

    domains: [
      {
        domain: 'myDomain.com',
        defaultLocale: 'en',
      },
      {
        domain: 'myDomain.es',
        defaultLocale: 'es',
      },
    ],
  },
}
```
#641
